### DISCUSSION TITLE
Dependent query disable the query cache

### DISCUSSION THREAD

**Discussion by bodinsamuel** [TanStack/query#4911] [Q&A] [answered]
Hey everyone,

I have been struggling the last few weeks with dependent queries. I'm very glad using it with react and the automatic caching of everything but it suddenly stops working when using dependent query.

When I have one `useQuery` with a a dynamic `queryKey`, and a second one that uses the first data. 
The second one is always refetched even though the queryKey is still the same.
Repro here: https://codesandbox.io/s/naughty-leaf-plxy58?file=/src/App.tsx

I might have completely missed something but I have read the doc and issues and didn't find this exact issue. 
NB: it does not happen if the first useQuery has a static queryKey.

> excerpt from codesandbox for searchability
```ts
  // We set the user after a timeout to simulate a query
  const user = useQuery({
    // the key here change to trigger a new fetch each time
    // Work as expected
    queryKey: ["getUser", amount],
    queryFn: async () => {
      return new Promise((resolve) => {
        setTimeout(() => {
          console.log(`getting user "${amount}" (expected)`);

          resolve({ id: "foobar" });
        }, 250);
      });
    }
  });

  // We wait for user to execute this query
  const something = useQuery({
    enabled: user.isLoading ? false : true,
    // The key does not change
    // It will always refetch even though the queryKey is the same
    queryKey: ["getSomething", user.data?.id],
    queryFn: async () => {
      console.log(`getting something "${user.data?.id}" (not expected)`);
      return { done: "true" };
    }
  });
```

Thanks,

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

Thanks for providing a reproduction. 

The issue is that query instances via `useQuery` or `useInfiniteQuery` by default consider cached data as stale. To change this behavior, you can configure your queries both globally and per-query using the `staleTime` option. Specifying a longer `staleTime` means queries will not re-fetch their data as often.

Please see: https://react-query-v3.tanstack.com/guides/important-defaults

Here is a forked fixed sandbox:

https://codesandbox.io/s/unruffled-fire-th10d1

**Reply by bodinsamuel** [reply]
Oh okay, thanks a lot for the quick answer.

I guess the combinaison of global options and local options makes it harder to predict the behavior, but as stated in your document it might be a problem of comprehension instead of an actual issue. 

Thanks again,
